sotodlib/toast/ops/dark_template.py

```diff
@@ -314,6 +314,11 @@ def _exec(self, data, detectors=None, **kwargs):
                 os.makedirs(self.cache_dir, exist_ok=True)
 
         for ob in data.obs:
+            if ob.dist.comm_col_size != 1:
+                raise RuntimeError(
+                    "DarkTemplate only works with observations distributed "
+                    "by detector"
+                )
```

Comment:
This is actually the opposite direction of what you want.  The column size is the size along each column (usually number of processes in the group) . The row size is usually 1.